src/controllers/suggestions.js

```diff
@@ -382,6 +386,96 @@ function SuggestionsController(dataAccess) {
     };
     return createResponse(fullResponse, 207);
   };
+  const autofixSuggestions = async (context) => {
+    const siteId = context.params?.siteId;
+    const opportunityId = context.params?.opportunityId;
+
+    if (!isValidUUID(siteId)) {
+      return badRequest('Site ID required');
+    }
+
+    if (!isValidUUID(opportunityId)) {
+      return badRequest('Opportunity ID required');
+    }
+
+    // validate request body
+    if (!context.data) {
+      return badRequest('No updates provided');
+    }
+    const { suggestionIds } = context.data;
+    if (!isArray(suggestionIds)) {
+      return badRequest('Request body must be an array of suggestionIds');
+    }
+    const site = await Site.findById(siteId);
+    if (!site) {
+      return notFound('Site not found');
+    }
+    const opportunity = await Opportunity.findById(opportunityId);
+    if (!opportunity || opportunity.getSiteId() !== siteId) {
+      return notFound('Opportunity not found');
+    }
+    const configuration = await Configuration.findLatest();
+    if (!configuration.isHandlerEnabledForSite(`${opportunity.getType()}-autofix`, site)) {
+      return badRequest(`Handler is not enabled for site ${site.getId()} autofix type ${opportunity.getType()}`);
+    }
+    const suggestionEntities = await Promise.all(suggestionIds.map(
+      async (suggestionId, index) => {
+        const suggestion = await Suggestion.findById(suggestionId);
```

Comment:
this would cause a separate DB query for each suggestion id. are all these suggestion. we could do `opportunity.getSuggestionsByStatus(NEW)` and `opportunity.getSuggestionsByStatus(IN_PROGRESS)`, then filter by the list of IDs...